tests/suites/test_suite_ssl.function

```diff
@@ -6237,10 +6237,21 @@ void ssl_ecjpake_set_password( int use_opaque_arg )
     {
         psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
 
-        psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_DERIVE );
+        /* First try with an invalid usage */
+        psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_SIGN_HASH );
         psa_set_key_algorithm( &attributes, PSA_ALG_JPAKE );
         psa_set_key_type( &attributes, PSA_KEY_TYPE_PASSWORD );
 
+        PSA_ASSERT( psa_import_key( &attributes, pwd_string,
+                    pwd_len, &pwd_slot ) );
+
+        ECJPAKE_TEST_SET_PASSWORD( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
+
+        psa_destroy_key( pwd_slot );
+
+        /* Then set the correct usage */
+        psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_DERIVE );
+
```

Comment:
There were two reasons for this test:

1. failing `psa_pake_set_password_key` (added);
2. checking if in case of a failure the same opaque key can be reused, so - if the key is not removed accidentaly by the newly added functions. This is unfortunately not satisfied, since you destroy the key and import it again. Is there a different possibility to make `psa_pake_set_password_key` that will let us reuse the same key without destroying it?

Reply:
Again, I don't think we need to re-use the key in order to make sure it was not destroyed: just query the keys attributes to get its type or algorithm - if the key has been destroyed this will not yield the expected result. We could add such a check at line 6249 above.

Reply:
Sure, that's also a way to handle it :)